packages/toolkit/src/codewhisperer/commands/startTransformByQ.ts

```diff
@@ -141,6 +157,47 @@ async function pickSourceVersion(
         transformByQState.setSourceJDKVersion(JDKVersion.JDK8)
     } else if (pick === validSourceVersions[1]) {
         transformByQState.setSourceJDKVersion(JDKVersion.JDK11)
+    } else if (pick === validSourceVersions[2]) {
+        // corresponds with the 'Other' option
+        transformByQState.setSourceJDKVersion(JDKVersion.UNSUPPORTED)
+    }
+}
+
+async function setMaven() {
+    let mavenWrapperExecutableName = os.platform() === 'win32' ? 'mvnw.cmd' : 'mvnw'
+    const mavenWrapperExecutablePath = path.join(transformByQState.getProjectPath(), mavenWrapperExecutableName)
+    if (fs.existsSync(mavenWrapperExecutablePath)) {
+        if (mavenWrapperExecutableName === 'mvnw') {
+            mavenWrapperExecutableName = './mvnw' // add the './' for non-Windows
+        }
+        transformByQState.setMavenName(mavenWrapperExecutableName)
+    } else {
+        transformByQState.setMavenName('mvn')
+    }
+}
+
+async function validateJavaHome() {
+    const versionData = await getVersionData()
+    let javaVersionUsedByMaven = versionData[1]
+    if (javaVersionUsedByMaven !== undefined) {
+        javaVersionUsedByMaven = javaVersionUsedByMaven.slice(0, 3)
+        if (javaVersionUsedByMaven === '1.8') {
+            javaVersionUsedByMaven = JDKVersion.JDK8
+        } else if (javaVersionUsedByMaven === '11.') {
+            javaVersionUsedByMaven = JDKVersion.JDK11
+        }
+    }
+    if (javaVersionUsedByMaven !== transformByQState.getSourceJDKVersion()) {
+        // means either javaVersionUsedByMaven is undefined or it does not match the project JDK
+        // TO-DO: give user help on how to find JAVA_HOME for corresponding project JDK
```

Comment:
Most likely this help will come in the form of a link to our documentation or some TBD messaging right in the IDE.